dvc/api.py

```diff
@@ -74,10 +80,14 @@ def __getattr__(self, name):
 
 def _open(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
     with _make_repo(repo, rev=rev) as _repo:
-        with _repo.open_by_relpath(
-            path, remote=remote, mode=mode, encoding=encoding
-        ) as fd:
-            yield fd
+        tree = RepoTree(_repo, stream=True, fetch=True)
+
+        path = PathInfo(_repo.root_dir) / path
+        with reraise(FileNotFoundError, PathMissingError(path, repo)):
+            with tree.open(
+                path, remote=remote, mode=mode, encoding=encoding
+            ) as fd:
+                yield fd
```

Comment:
Doesn't look like there is a need for this change. Repo.open_by_relpath already does all of this.

Reply:
`open_by_relpath` will most likely go away, as `RepoTree` supports subrepos directly.

Reply:
@skshetry Agreed, it was only used for the API, so we could delete it in that case.

Reply:
Exceptions do not look good, I do have to tell you that because `Repo` and `ExternalRepo` threw different exception before, but now, I am throwing `PathMissingError`. Unified, but quite verbose.